### PR TITLE
Introduce BaseURL attribute

### PR DESCRIPTION
**Pull request by lzap**
SSIA

### CODE REVIEW

**Comment by lzap** [comment]
Let me integrate link checker into this PR, give me 5 minutes.

**Comment by lzap** [comment]
So I am adding linkchecker too, please `yum install linkchecker` in your fedora and then run `make linkchecker` to see it. We actually have one incorrect link which is "see Creating a Role."

I have added this into Travis so this PR will now should fail. Investigate the Travis result to learn how to work with the tool and once you are comfortable with that I can fix it.

**Comment by lzap** [comment]
As you can see, travis failed with two errors:

```
URL        `https://access.redhat.com/documentation/en-us/red_hat_satellite/6.4/html/administering_red_hat_satellite/chap-Foreman-Administering_Foreman-Users_and_Roles#sect-Foreman-Administering_Foreman-Creating_and_Managing_Roles-Creating_a_Role'

Name       `Creating a Role'

Parent URL file:///home/travis/build/theforeman/foreman-documentation/guides/build/doc-Provisioning_Guide/index-foreman.html, line 3757, col 51

Real URL   https://access.redhat.com/documentation/en-us/red_hat_satellite/6.4/html/administering_red_hat_satellite/chap-Foreman-Administering_Foreman-Users_and_Roles#sect-Foreman-Administering_Foreman-Creating_and_Managing_Roles-Creating_a_Role

Check time 2.414 seconds

Result     Error: 404 Not Found

URL        `https://access.redhat.com/documentation/en-us/red_hat_satellite/6.4/html/administering_red_hat_satellite/chap-Foreman-Administering_Foreman-Users_and_Roles#sect-Foreman-Administering_Foreman-Creating_and_Managing_Roles-Adding_Permissions_to_a_Role'

Name       `Adding Permissions to a Role'

Parent URL file:///home/travis/build/theforeman/foreman-documentation/guides/build/doc-Provisioning_Guide/index-foreman.html, line 3757, col 378

Real URL   https://access.redhat.com/documentation/en-us/red_hat_satellite/6.4/html/administering_red_hat_satellite/chap-Foreman-Administering_Foreman-Users_and_Roles#sect-Foreman-Administering_Foreman-Creating_and_Managing_Roles-Adding_Permissions_to_a_Role

Check time 2.629 seconds

Result     Error: 404 Not Found
```

Going to fix them in another commit in this PR.

**Comment by lzap** [comment]
Ok the reason why it was broken was that I actually replaced "Red_Hat_Satellite" with the attribute ignoring case sensitivity. It replaced these two guys as well which was my bad. The script will do it properly matching the case.

Also it revealed one more issue - we can only set the Project_Link attribute to Foreman only after we move everything into our new docs server, until then we are stuck with downstream anchors.

**Comment by melcorr** [comment]
> Ok the reason why it was broken was that I actually replaced "Red_Hat_Satellite" with the attribute ignoring case sensitivity. It replaced these two guys as well which was my bad. The script will do it properly matching the case.
> 
> Also it revealed one more issue - we can only set the Project_Link attribute to Foreman only after we move everything into our new docs server, until then we are stuck with downstream anchors.

Ah!

**Comment by lzap** [comment]
It's now green.

**Comment by melcorr** [comment]
I am just building locally to validate these changes

**Comment by melcorr** [comment]
I don't know if it was introduced in this PR, but this whole section is broken: 

7.1. Prerequisites for KVM Provisioning

The requirements for KVM provisioning include:

**Comment by melcorr** [comment]
> I don't know if it was introduced in this PR, but this whole section is broken:
> 
> 7.1. Prerequisites for KVM Provisioning
> 
> The requirements for KVM provisioning include:

All looks good so I will push a fix for this now.

File: guides/doc-Provisioning_Guide/topics/Virt-Libvirt.adoc [hunk 1]
```diff
@@ -8,21 +8,21 @@ Kernel-based Virtual Machines (KVMs) use an open source virtualization daemon an
 
 The requirements for KVM provisioning include:
 
-	ifeval::["{build}" == "satellite"]
-  * Synchronized content repositories for the version of Red Hat Enterprise Linux that you want to use. For more information, see {BaseURL}content_management_guide/importing_red_hat_content#Importing_Red_Hat_Content-Synchronizing_Red_Hat_Repositories[Synchronizing Red Hat Repositories] in the _Content Management Guide_.
-	endif::[]
-	ifeval::["{build}" == "foreman"]
+ifeval::["{build}" == "satellite"]
+  * Synchronized content repositories for the version of Red Hat Enterprise Linux that you want to use. For more information, see link:/html/content_management_guide/importing_red_hat_content#Importing_Red_Hat_Content-Synchronizing_Red_Hat_Repositories[Synchronizing Red Hat Repositories] in the _Content Management Guide_.
```
Comment (lzap) [inline guides/doc-Provisioning_Guide/topics/Virt-Libvirt.adoc]: You replaced BaseURL back with `link:/html` which will not work.

**Comment by melcorr** [comment]
@lzap, I reviewed everything as if this was the last PR of this guide. I fixed a few things. 
I found a few references to Capsule, updated those. 
I standardized a step about the activation key with an ifeval

I then found that this file: app_create_images.adoc has not been reviewed to the same standard as the other topics. I started to review it, but then I got very confused as to whether we could just omit this altogether for Foreman. I remember when this content was added, and it was at the request of a customer. It probably needs updating, but I doubt it has any relevance to Foreman. 

What do you think?

**Comment by melcorr** [comment]
I added an ifeval around that entire topic. Perhaps that will do for the moment and we can work on whether this content really applies upstream? 

Let me know what you think.

**Comment by lzap** [comment]
I've reviewed everything, I agree it's better to hide this appendix until we deliver something reasonable. I suggest to break it into assemblies: rhel6, rhel7 and fedora. We would ship only fedora one upstream.

**Comment by lzap** [comment]
Created a ticket for that, merging this one then: https://github.com/theforeman/foreman-documentation/issues/52

**Pull request merged**